esp-link/config.c

```diff
@@ -9,21 +9,31 @@
 
 FlashConfig flashConfig;
 FlashConfig flashDefault = {
-  33, 0, 0,
-  MCU_RESET_PIN, MCU_ISP_PIN, LED_CONN_PIN, LED_SERIAL_PIN,
-  115200,
-  "esp-link\0",                        // hostname
-  0, 0x00ffffff, 0,                    // static ip, netmask, gateway
-  0,                                   // log mode
-  0,                                   // swap uart (don't by default)
-  1, 0,                                // tcp_enable, rssi_enable
-  "\0",                                // api_key
-  0, 0, 0,                             // slip_enable, mqtt_enable, mqtt_status_enable
-  2, 1,                                // mqtt_timeout, mqtt_clean_session
-  1883, 60,                            // mqtt port, mqtt_keepalive
-  "\0", "\0", "\0", "\0", "\0",        // mqtt host, client_id, user, password, status-topic
-  "\0",                                // system description
-  1,                                   // rx_pullup
+  .seq=33, .magic=0, .crc=0,
+  .reset_pin    = MCU_RESET_PIN, .isp_pin      = MCU_ISP_PIN,
+  .conn_led_pin = LED_CONN_PIN,  .ser_led_pin  = LED_SERIAL_PIN,
+  .baud_rate    = 115200,
+  .hostname     = "ems-link02\0",
```

Comment:
You mean the designated initializer per se or my (accidently promoted) 
hostname?

On 11/24/2015 06:45 AM, Thorsten von Eicken wrote:
>
> In esp-link/config.c 
> <https://github.com/jeelabs/esp-link/pull/73#discussion_r45698825>:
>
> > -  0, 0x00ffffff, 0,                    // static ip, netmask, gateway
> > -  0,                                   // log mode
> > -  0,                                   // swap uart (don't by default)
> > -  1, 0,                                // tcp_enable, rssi_enable
> > -  "\0",                                // api_key
> > -  0, 0, 0,                             // slip_enable, mqtt_enable, mqtt_status_enable
> > -  2, 1,                                // mqtt_timeout, mqtt_clean_session
> > -  1883, 60,                            // mqtt port, mqtt_keepalive
> > -  "\0", "\0", "\0", "\0", "\0",        // mqtt host, client_id, user, password, status-topic
> > -  "\0",                                // system description
> > -  1,                                   // rx_pullup
> > +  .seq=33, .magic=0, .crc=0,
> > +  .reset_pin    = MCU_RESET_PIN, .isp_pin      = MCU_ISP_PIN,
> > +  .conn_led_pin = LED_CONN_PIN,  .ser_led_pin  = LED_SERIAL_PIN,
> > +  .baud_rate    = 115200,
> > +  .hostname     = "ems-link02\0",
>
> I don't think this makes sense as a default
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jeelabs/esp-link/pull/73/files#r45698825>.
>